neovim: 0.2.2 -> 0.3.0

teto opened the pull request:
Changelog at https://github.com/neovim/neovim/commit/44c6deb91ac917703c157d564eb3accbff4d37af

Fixed missing manpage.
Remove the indirection for the python interpreters (why was it there in the first place ? I am afraid there is a good reason but I don't know it).

Sadly we still can't run the tests but if anyone is interested, this used to be possible with
https://github.com/NixOS/nixpkgs/pull/33903#pullrequestreview-97792751

###### Motivation for this change

Neovim is freaking cool

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

xeji commented:
@grahamcofborg build neovim

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41922&attempt_id=220dbe74-7a41-41ec-9c83-6f03fc309c88)

Attempted: neovim

<details><summary>Partial log (click to expand)</summary><p>

```
these derivations will be built:
  /nix/store/r8w78gi8x5144ns5zlj8zn8iw48wk3my-neovim-0.3.0.drv
building '/nix/store/r8w78gi8x5144ns5zlj8zn8iw48wk3my-neovim-0.3.0.drv'...
/nix/store/60zkc8xm499147v33wj9yfw3gvarxkkm-neovim-0.3.0
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41922&attempt_id=8a985845-6bc8-430e-9109-894fb477b7c6)

Attempted: neovim

<details><summary>Partial log (click to expand)</summary><p>

```
stack traceback:
        [C]: in function 'match'
        /build/source/src/nvim/generators/gen_declarations.lua:255: in main chunk
        [C]: at 0x00405660
make[2]: *** [src/nvim/CMakeFiles/nvim.dir/build.make:235: src/nvim/auto/eval.c.generated.h] Error 1
make[1]: *** [CMakeFiles/Makefile2:2865: src/nvim/CMakeFiles/nvim.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/g76lcg5l56fza9jc4zdb46s088qq45a8-neovim-unwrapped-0.3.0.drv' failed with exit code 2
cannot build derivation '/nix/store/ix69iyynf6iis6kbmpnvx550plfjb0n0-neovim-0.3.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ix69iyynf6iis6kbmpnvx550plfjb0n0-neovim-0.3.0.drv' failed
```
</p></details>

xeji commented:
neovim was already upgraded to 0.3.0 in #41873. Closing this as duplicate.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41922&attempt_id=054baf92-2a20-4702-9b1a-ce5929d0c49c)

Attempted: neovim

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/5q3s7vgbx7ldnnh2q4mjnsnxkpi88gc9-python3.6-neovim-0.2.6' from 'https://cache.nixos.org'...
copying path '/nix/store/2803z50pzlzb8yww4aq834nv6pf3rlc4-ruby2.4.4-msgpack-1.2.4' from 'https://cache.nixos.org'...
copying path '/nix/store/84rc2fa3slz99ygsfavy7mdr5klysszf-python3-3.6.5-env' from 'https://cache.nixos.org'...
copying path '/nix/store/wv1jyd4ql89g8kf7y93wd4cnf2yzhsxp-ruby2.4.4-neovim-0.7.0' from 'https://cache.nixos.org'...
copying path '/nix/store/fxvisrcfka11633834djb88ls3r32xwy-unibilium-2.0.0' from 'https://cache.nixos.org'...
copying path '/nix/store/z3s4lqj2z8lbwgddaicscpv1lr3m7h1k-neovim-ruby-env' from 'https://cache.nixos.org'...
copying path '/nix/store/ikm2lkimfcbvaggzm1r2d5iaijj3g01y-xsel-unstable-2016-09-02' from 'https://cache.nixos.org'...
copying path '/nix/store/xr6hr01ym9adgj7hjicjhngingbfdwr6-neovim-unwrapped-0.3.0' from 'https://cache.nixos.org'...
building '/nix/store/lnym5fxpxg1v53llpi34y1g7pn0q89jx-neovim-0.3.0.drv'...
/nix/store/wv82s7caxz2z50fis5k5c6q0kskwjalg-neovim-0.3.0
```
</p></details>

teto committed:
neovim: 0.2.2 -> 0.3.0

Changelog at https://github.com/neovim/neovim/commit/44c6deb91ac917703c157d564eb3accbff4d37af

Fixed missing manpage.
Remove the indirection for the python interpreters
